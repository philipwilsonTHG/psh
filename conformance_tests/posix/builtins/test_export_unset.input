# Test export and unset builtins

# Test basic export
export TEST_VAR="test value"
echo "Exported: $TEST_VAR"

# Test export without assignment
TEST_VAR2="another value"
export TEST_VAR2
echo "Exported later: $TEST_VAR2"

# Test showing exported variables (limit output)
export | grep TEST_VAR | head -2

# Test unset
unset TEST_VAR
echo "After unset: $TEST_VAR"

# Test unset with export
unset TEST_VAR2
export | grep TEST_VAR2 || echo "TEST_VAR2 not found in exports"

# Test export with -p flag
export -p | grep TEST_VAR || echo "No TEST_VAR variables found"