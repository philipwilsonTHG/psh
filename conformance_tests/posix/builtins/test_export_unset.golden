Exported: test value
Exported later: another value
After unset:
TEST_VAR2 not found in exports
No TEST_VAR variables found
